File: cpp/make_max_num1.cpp
// 최댓값 만들기 1

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

/* Answer 1

int solution(vector<int> numbers) {
    int answer = 0;
    int vec_size = 0;
    
    sort(numbers.begin(), numbers.end());
    vec_size = numbers.size();
    answer = numbers[vec_size - 1] * numbers[vec_size - 2];
    
    return answer;
} */


/* Answer 2

int solution(vector<int> numbers) {
    int answer = 0;
    
    sort(numbers.begin(), numbers.end());
    reverse(numbers.begin(), numbers.end());
    answer = numbers[0] * numbers[1];
    
    return answer;
} */


/* Answer 3

int solution(vector<int> numbers) {
    int answer = 0;
    
    sort(numbers.begin(), numbers.end());
    sort(numbers.rbegin(), numbers.rend());
    answer = numbers[0] * numbers[1];
  
    return answer;
} */

File: python/lv0/make_max_num1.py
# 최댓값 만들기(1)

'''
Answer 1

def solution(numbers):
    answer = 0
    
    numbers.sort()
    max_num = numbers[len(numbers) - 1]
    second_num = numbers[len(numbers) - 2]
    answer = max_num * second_num
    
    return answer

'''

'''
Answer 2

def solution(numbers):
    numbers.sort(reverse=True)
    
    return numbers[0]*numbers[1]

'''

'''
Answer 3

def solution(numbers):
    numbers.sort()
    return numbers[-2] * numbers[-1]

'''

'''
Answer 4

def solution(numbers):
    return sorted(numbers)[-1] * sorted(numbers)[-2]

'''
